lib/services/irc.rb

```diff
@@ -137,6 +137,31 @@ def irc_puts(command, debug_command=command)
     writable_irc.puts command
   end
 
+  def irc_realname
+    repo_owner = payload["repository"]["owner"]["name"]
+    repo_name = payload["repository"]["name"]
+
+    "GitHub IRCBot - #{repo_owner}" + (repo_public? ? "/#{repo_name}" : "")
+  end
+
+  def repo_public?
```

Comment:
> Do you all think the `private` field will exist for all events that contain a `repository`?

Yes. You can see all the possible event types [here](https://github.com/github/github-services/blob/2e886f407696261bd5adfc99b16d36d5e7b50241/lib/service.rb#L79-L83). And you can see an example payload for each event type [here](https://developer.github.com/v3/activity/events/types/).

> I couldn't find any other service doing a similar check, and the test payloads did not seem to contain them.

It looks like the test payloads are incomplete. I recommend adding the `private` field to the test payload to support the the tests for this pull request. That will eliminate the need to make an API request to determine whether the repository is public or private.